### DISCUSSION TITLE
stunnel always says TLS server needs a certificate

### DISCUSSION THREAD

**Discussion by jintor** [Homebrew/discussions#3113] [Everyday usage] [answered]
I'm unable to get stunnel to work.

I did brew install stunnel

in /usr/local/etc/stunnel/stunnel.conf
```
[https]
client = yes
accept  = 82
connect = 81
cert = /usr/local/etc/stunnel/stunnel.pem
TIMEOUTclose = 0
sslVersion = all
```

in the terminal I do `stunnel3`
[ ] Initializing inetd mode configuration
[ ] Clients allowed=125
[.] stunnel 5.63 on x86_64-apple-darwin19.6.0 platform
[.] Compiled/running with OpenSSL 3.0.2 15 Mar 2022
[.] Threading:PTHREAD Sockets:POLL,IPv6 TLS:ENGINE,OCSP,PSK,SNI
[ ] errno: (*__error())
[ ] Initializing inetd mode configuration
[.] Reading configuration from descriptor 3
[.] FIPS mode disabled
[ ] Compression enabled: 0 methods
[ ] No PRNG seeding was required
[!] Inetd mode: TLS server needs a certificate
[!] Configuration failed
[ ] Deallocating temporary section defaults

*** HOW do I fix the [!] Inetd mode: TLS server needs a certificate ????

I have mac os 10.15

**Comment by SMillerDev** [comment]
This seems like a question that the stunnel community would be better suited to answer but I'll give it a try. Does the certificate you specified actually exist?

**Reply by jintor** [reply]
yes, when I cat /usr/local/etc/stunnel/stunnel.pem 

I get the -----BEGIN PRIVATE KEY----- -----BEGIN CERTIFICATE----- stuff

no matter what I try, I always get "TLS server needs a certificate"

**Comment by jintor** [comment, marked as answer]
I fixed it with this command

stunnel /pathtowhereisconfigfile/stunnel.conf

then it worked

I also needed to separate the .pem (generated by the stunnel install) into a .crt and .key and in config file

cert = /etc/stunnel/stunnel.crt
key = /etc/stunnel/stunnel.key
